Update documentation for loacal developement on bare metal

browniebroke opened the pull request:
I'm hoping that this update will clarify the bare metal setup which is causing recurring issues. This is essentially adding some details to the local development on bare metal, but also fixes some outdated settings and a couple of minor styling issues.

Refs #1781 
Refs #1773 
Refs #1775 
Fixes #564

browniebroke commented on docs/developing-locally.rst:
```diff
@@ -9,24 +9,53 @@ Setting Up Development Environment
+
+       if this is the first time a database is created on your machine you might need to alter
+       a localhost-related entry in your ``pg_hba.conf`` so as to utilize ``trust`` policy, see the
+       `postgres documentation`_ for more details.
```
I'm not 100% sure what this means tbh, I never had to set this up on my machine. If anyone has more details, I'd be more than happy to elaborate.

webyneter replied:
I believe this was my input: in past, I did face the issue with `pg_hba.conf` localhost auth records.

webyneter replied:
I suggest that this note stays intact: it proved itself helpful to me in past, no reason to assume the opposite for everyone else. It might require clarifications, though.

luzfcb replied:
I usually set up password for the `postgres` database user and change the authentication mode to md5:

in Ubuntu would be something like this:

```bash
# runs the "psql" command as a "postgres" linux user, and changes the password of "postgres" database user
# yes, when you install postgres on linux (at least on Ubuntu), a linux user named "postgres" is created

sudo -u postgres psql postgres -c "ALTER USER postgres WITH PASSWORD 'new_password';"
```
change `pg_hba.conf` configuration file and enable md5 authentication

>         in ubuntu, this file is located in:
>        /etc/postgresql/POSTGRESH_VERSION_NUMBER/main/pg_hba.conf
>        
>        if the version of postgres is 9.6
>
>       
>        /etc/postgresql/9.6/main/pg_hba.conf
>        
>        if is 10
>        
>        /etc/postgresql/10/main/pg_hba.conf


open and change the pg_hba.conf
from:
```
local   all             all                                     peer
```
to
```
local   all             all                                     md5
```
restart postgresql

```
sudo service postgresql restart

```

luzfcb replied:
here is a simple, and good documentation on how to configure postgresql for first use

http://suite.opengeo.org/docs/latest/dataadmin/pgGettingStarted/firstconnect.html

browniebroke replied:
I'd really like to add more details on this, but it seems to be a bit overwhelming to include in this page.

I've looked at this to see how different it is on macOS. I've re-installed `postgresql` with Homebrew and it looks like the setup is a bit different. The `pg_hba.conf` file is located under `/usr/local/var/postgres/pg_hba.conf` and it uses the `trust` policy. The DB server is running as the current user (not postgres), but has no password either.

Finally, I have no way to test how this would work under Windows...

I've deleted `docs/installing_postgres.rst` for now, I could either add it back and elaborate on this topic, or link to that external documentation.

browniebroke commented on docs/developing-locally.rst:
```diff
@@ -9,24 +9,53 @@ Setting Up Development Environment
 Make sure to have the following on your host:
 
-* virtualenv_;
-* pip;
```
I dropped `pip` & `virtualenv` as they both come pre-installed with Python 3.6.

browniebroke commented on docs/developing-locally.rst:
```diff
@@ -9,24 +9,53 @@ Setting Up Development Environment
+#. Create a virtualenv: ::
 
-#. Activate the virtualenv you have just created.
+    $ python3 -m venv <virtual env path>
```
I just realised that `python3` might point to Python 3.7, which isn't supported yet (due to Celery, at least).

sfdye replied:
why not use `virtualenv <path/to/new/virtualenv/> --python=python3.6`

browniebroke replied:
This one is included in [python since version 3.3](https://docs.python.org/3.6/library/venv.html), so that's one fewer thing to install, no need to have a requirement on the standalone `virtualenv` library.

Correct me if I'm wrong, but I think they do the same thing.

sfdye replied:
I think you are right. We can get rid of virtualenv. They basically do the same thing but is `venv` is a built-in module.

https://www.reddit.com/r/learnpython/comments/4hsudz/pyvenv_vs_virtualenv/

So we can use `python3.6 -m <virtual env path>` or `/path/to/python3.6 -m <virtual env path>`

luzfcb commented on docs/developing-locally.rst:
```diff
@@ -9,24 +9,53 @@ Setting Up Development Environment
+
+#. Set the environment variables for your database(s): ::
+
+    $ export DATABASE_URL=postgres://<project_slug>
```
if use the way I configure the postgres, then must also pass the user and password as a parameter, I'm also explicit about the host and port

```
export DATABASE_URL=postgres://postgres:password@127.0.0.1:5432
```

browniebroke replied:
Agreed. I think it would be better to include the entire thing here, actually:

```
export DATABASE_URL=postgres://<user>:<password>@127.0.0.1:5432/<db_name>
```

browniebroke commented:
I think I have addressed all the comments. This is still perfectible, but it's better than the current version IMO, so I'd say it's worth merging. We can improve further from there. Thoughts?
